card/src/main/java/com/adyen/checkout/card/internal/ui/DefaultCardDelegate.kt

```diff
@@ -233,6 +236,9 @@ internal class DefaultCardDelegate(
                     "New detected card types emitted - detectedCardTypes: ${detectedCardTypes.map { it.cardBrand }} " +
                         "- isReliable: ${detectedCardTypes.firstOrNull()?.isReliable}"
                 )
+                if (outputData.detectedCardTypes != detectedCardTypes) {
```

Comment:
Can we do 
```
detectCardTypeRepository.detectedCardTypesFlow
            .distinctUntilChanged()
            .onEach { detectedCardTypes ->
```
instead of having this if? I think it wouldn't change the behaviour. It would get triggered fewer times but should be fine. What do you think?

Reply:
It does actually change the behaviour, because currently we update the output data even when the detected card types are the same as before. I like your suggestion though, because it doesn't make sense to update the output data if nothing changed